corehq/apps/users/views/mobile/users.py

```diff
@@ -1313,6 +1314,41 @@ def _delete_users(self, request, user_docs_by_id, user_ids_with_forms):
             messages.success(request, f"{deleted_count} user(s) deleted.")
 
 
+@method_decorator([toggles.CLEAR_MOBILE_WORKER_DATA.required_decorator()], name='dispatch')
+class ClearCommCareUsers(DeleteCommCareUsers):
+    urlname = 'clear_commcare_users'
+    page_title = gettext_noop('Bulk Clear')
+    template_name = 'users/bulk_clear.html'
+
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        usernames = self._get_usernames(request)
+        if not usernames:
+            return self.get(request, *args, **kwargs)
+
+        user_docs_by_id = {doc['_id']: doc for doc in get_user_docs_by_username(usernames)}
+        usernames_not_found = self._get_usernames_not_found(request, user_docs_by_id, usernames)
+
+        if usernames_not_found:
+            messages.error(request, _("""
+                No users cleared. Please address the above issue(s) and re-upload your updated file.
```

Comment:
Should we cancel the entire clear process if there are still valid usernames? From the error message it seems that this is for when no users could be cleared. In that case, should we not then check `if len(usernames_not_found) == len(usernames)`?

Reply:
That did cross my mind, but I modelled this after the current bulk delete functionality.